### PR TITLE
[12][MIG] base optional quick create mgs

### PR DESCRIPTION
**Pull request by GillesTephaneMeyomesse**
#1368

### CODE REVIEW

File: base_optional_quick_create/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# Copyright 2013 Agile Business Group sagl (<http://www.agilebg.com>)
+# Copyright 2016 ACSONE SA/NV (<https://acsone.eu>)
+
+{
+    'name': 'Optional quick create',
+    'version': '12.0.1.0.0',
+    'category': 'Tools',
+    'summary': 'Avoid "quick create" on m2o fields, on a "by model" basis',
+    'author': 'Agile Business Group,Odoo Community Association (OCA)',
+    'website': 'http://www.agilebg.com',
```
Comment (acsonefho) [inline base_optional_quick_create/__manifest__.py]: Url should be replace by this [one](https://github.com/OCA/maintainer-tools/blob/master/template/module/__manifest__.py#L10).

File: base_optional_quick_create/models/ir_model.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+    def _register_hook(self):
+        models = self.search([])
+        models._patch_quick_create()
+        return super(IrModel, self)._register_hook()
+
+    @api.model
+    @api.returns('self', lambda value: value.id)
+    def create(self, vals):
+        ir_model = super(IrModel, self).create(vals)
```
Comment (acsonefho) [inline base_optional_quick_create/models/ir_model.py]: In Python 3 the super should be empty:
`ir_model = super().create(vals)`

File: base_optional_quick_create/models/ir_model.py [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+        ir_model._patch_quick_create()
+        return ir_model
+
+    @api.multi
+    def write(self, vals):
+        res = super(IrModel, self).write(vals)
```
Comment (acsonefho) [inline base_optional_quick_create/models/ir_model.py]: Same here (about `super()`)

File: base_optional_quick_create/tests/test_quick_create.py [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo.exceptions import UserError
+from odoo.tests.common import TransactionCase
+
+
+class TestQuickCreate(TransactionCase):
+
+    def setUp(self, *args, **kwargs):
+        super(TestQuickCreate, self).setUp()
```
Comment (acsonefho) [inline base_optional_quick_create/tests/test_quick_create.py]: Same here (about `super()`)

**Review by acsonefho — CHANGES_REQUESTED** [review]
Minor changes. Otherwise code looks good

**Comment by pedrobaeza** [comment]
This should go to OCA/server-ux instead of here.

**Comment by GillesTephaneMeyomesse** [comment]
@pedrobaeza Ok, I'll change it

**Comment by GillesTephaneMeyomesse** [comment]
@pedrobaeza done https://github.com/OCA/server-ux/pull/31

**Pull request closed without merging**